apps/prairielearn/assets/scripts/lib/saveButtonEnabling.ts

```diff
@@ -28,12 +36,27 @@ export function saveButtonEnabling(form: HTMLFormElement, saveButton: HTMLButton
   // Similar to the above, but for select elements. The difference here being that select elements do not store the default value so we must store those in selectDefaultValues{} and compare against those.
 
   form.addEventListener('change', (e) => {
-    if (
-      (e.target as HTMLInputElement).value === selectDefaultValues[(e.target as HTMLElement).id]
-    ) {
-      valueHasChanged[(e.target as HTMLElement).id] = false;
+    if ((e.target as HTMLInputElement).multiple) {
```

Comment:
The types here are a bit confused; you treat `e.target` as an `HTMLInputElement` in this conditional, and then in the following block, treat it as an `HTMLSelectElement`.

Given that you expect this callback to only do anything for `<select>` inputs, you can simplify this quite a bit with an early return and some other changes:

```ts
    if (!(e.target instanceof HTMLSelectElement)) return;

    if (e.target.multiple) {
      // If the select element is a multiple select, we must compare the selected options to the default values.
      const selectedOptions = Array.from(e.target.selectedOptions).map((option) => option.value);

      valueHasChanged[e.target.id] =
        JSON.stringify(selectedOptions.sort()) !==
        JSON.stringify(selectDefaultValues[e.target.id].sort());
    } else {
      valueHasChanged[e.target.id] = e.target.value !== selectDefaultValues[e.target.id];
    }
```

In fact, because `selectedOptions` is _still_ a property even for single-option `<select>` elements, you can simplify it even further:

```ts
    if (!(e.target instanceof HTMLSelectElement)) return;

    const selectedOptions = Array.from(e.target.selectedOptions).map((option) => option.value);

    valueHasChanged[e.target.id] =
      JSON.stringify(selectedOptions.sort()) !==
      JSON.stringify(selectDefaultValues[e.target.id].sort());
```

This will also require you to change the initialization of `selectDefaultValues` in a similar way (treating both single-select and multiple-select the same way).

Reply:
That is much cleaner. Updated this and `selectDefaultValues` in [d426d26](https://github.com/PrairieLearn/PrairieLearn/pull/10973/commits/d426d26b01edba612257c99691275d279950c468).